### PR TITLE
Added log4j2 FileLogger

### PR DESCRIPTION
**Pull request by Grigory-Rylov**

### CODE REVIEW

File: src/main/java/com/github/grishberg/tests/common/FileLogger.kt [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    override fun w(tag: String, message: String) {
+        logger.warn("$tag : $message")
+    }
+
+    override fun i(tag: String, message: String) {
+        logger.info("$tag : $message")
+    }
+
+    override fun i(tag: String, msgFormat: String, vararg args: Any?) {
+        logger.info("$tag : $msgFormat", args)
```
Comment (Nexen23) [inline src/main/java/com/github/grishberg/tests/common/FileLogger.kt]: Shouldn't there be `String.format(msgFormat, args)` instead?

File: src/main/resources/log4j2.xml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+            <PatternLayout>
+                <pattern>%d{yyyy-MM-dd HH:mm:ss.SSS} [%-5level] [%t] - %msg%n</pattern>
+            </PatternLayout>
+            <Policies>
+                <TimeBasedTriggeringPolicy interval="1" modulate="true" />
+            </Policies>
+        </RollingFile>
+
+        <Console name="console" target="SYSTEM_OUT">
+            <PatternLayout   pattern="%d{yyyy-MM-dd HH:mm:ss.SSS} [%-5level] [%t] - %msg%n" />
```
Comment (Nexen23) [inline src/main/resources/log4j2.xml]: spaces

File: src/main/java/com/github/grishberg/tests/common/FileLogger.kt [hunk 3]
```diff
@@ -0,0 +1,39 @@
+package com.github.grishberg.tests.common
+
+import org.apache.logging.log4j.LogManager
+
+class FileLogger : RunnerLogger {
+    private val logger = LogManager.getLogger(FileLogger::class.java.name)
```
Comment (Nexen23) [inline src/main/java/com/github/grishberg/tests/common/FileLogger.kt]: `this::class.java.name`?

File: src/main/java/com/github/grishberg/tests/common/FileLogger.kt (hunk 1 above)
Comment (Grigory-Rylov) [reply src/main/java/com/github/grishberg/tests/common/FileLogger.kt]: No, shouldn't.  Runner used this format 👍 `"Before executing device = {} command = {}"`

there is sample logger output:
```
2019-03-02 22:36:28.764 [INFO ] [main] - InstrumentationTestTask : InstrumentationTestTask.runTask
2019-03-02 22:36:28.789 [INFO ] [main] - InstrumentationTestTask : androidSdkPath is empty, get path from env ANDROID_HOME
2019-03-02 22:36:28.791 [INFO ] [main] - InstrumentationTestTask : androidSdkPath = [/Users/grishberg/Library/Android/sdk]
2019-03-02 22:36:28.792 [INFO ] [main] - InstrumentationTestTask : Init: commandsForAnnotationProvider is empty, use DefaultCommandsForAnnotationProvider
2019-03-02 22:36:28.793 [INFO ] [main] - InstrumentationTestTask : init: instrumentationArgsProvider is empty, use DefaultInstrumentationArgsProvider
2019-03-02 22:36:28.793 [INFO ] [main] - InstrumentationTestTask : command provider is empty, use DefaultCommandProvider
2019-03-02 22:36:28.824 [DEBUG] [main] - InstrumentationTestTask : Reports dir is empty, generate default value [/private/var/folders/95/kpxvd2s53452jhdwwqj70d0r0000gn/T/gradle8517688036223972396projectDir/build/outputs/reports/androidTest/TEST_FLAVOR]
2019-03-02 22:36:28.831 [DEBUG] [main] - InstrumentationTestTask : Results dir is empty, generate default value [/private/var/folders/95/kpxvd2s53452jhdwwqj70d0r0000gn/T/gradle8517688036223972396projectDir/build/outputs/androidTest/TEST_FLAVOR]
2019-03-02 22:36:28.832 [DEBUG] [main] - InstrumentationTestTask : Coverage dir is empty, generate default value [/private/var/folders/95/kpxvd2s53452jhdwwqj70d0r0000gn/T/gradle8517688036223972396projectDir/build/outputs/androidTest/coverage/TEST_FLAVOR]
2019-03-02 22:36:29.843 [INFO ] [Thread-1] - DefaultCommandProvider : provideCommandsForDevice: device = [ConnectedDeviceWrapper{sn=emulator-5554, isOnline=true, name='Nexus_5X_API_27 [emulator-5554]'}, {}], args = {}
2019-03-02 22:36:30.236 [INFO ] [Thread-1] - DCR : Before executing device = [ConnectedDeviceWrapper{sn=emulator-5554, isOnline=true, name='Nexus_5X_API_27 [emulator-5554]'}, com.github.grishberg.tests.commands.SetAnimationSpeedCommand@74c33216] command = {}
2019-03-02 22:36:30.358 [INFO ] [Thread-1] - DCR : After executing device = [ConnectedDeviceWrapper{sn=emulator-5554, isOnline=true, name='Nexus_5X_API_27 [emulator-5554]'}, com.github.grishberg.tests.commands.SetAnimationSpeedCommand@74c33216] command = {}
2019-03-02 22:36:30.358 [INFO ] [Thread-1] - DCR : Before executing device = [ConnectedDeviceWrapper{sn=emulator-5554, isOnline=true, name='Nexus_5X_API_27 [emulator-5554]'}, com.github.grishberg.tests.commands.SetAnimationSpeedCommand@5924155b] command = {}
2019-03-02 22:36:30.483 [INFO ] [Thread-1] - DCR : After executing device = [ConnectedDeviceWrapper{sn=emulator-5554, isOnline=true, name='Nexus_5X_API_27 [emulator-5554]'}, com.github.grishberg.tests.commands.SetAnimationSpeedCommand@5924155b] command = {}
```

File: src/main/java/com/github/grishberg/tests/common/FileLogger.kt (hunk 3 above)
Comment (Grigory-Rylov) [reply src/main/java/com/github/grishberg/tests/common/FileLogger.kt]: Its fine, I hide logger class name from output

File: src/main/resources/log4j2.xml (hunk 2 above)
Comment (Grigory-Rylov) [reply src/main/resources/log4j2.xml]: fixed

**Pull request merged**